src/main/kotlin/com/atlassian/performance/tools/jiraactions/api/scenario/JiraAllActionScenario.kt

```diff
@@ -0,0 +1,130 @@
+        issueKeyMemory?.let { this.issueKeyMemory = it }
+    }
+
+    override fun getActions(jira: WebJira, seededRandom: SeededRandom, meter: ActionMeter): List<Action> {
+        initializeIssueKeyMemory(seededRandom)
+        val projectMemory = AdaptiveProjectMemory(random = seededRandom)
+        val jqlMemory = AdaptiveJqlMemory(seededRandom)
+        val issueMemory = AdaptiveIssueMemory(issueKeyMemory, seededRandom)
+
+        val scenario: MutableList<Action> = mutableListOf()
+
+        val createIssue = CreateIssueAction(
+            jira = jira,
+            meter = meter,
+            seededRandom = seededRandom,
+            projectMemory = projectMemory
+        )
+        val searchWithJql = SearchJqlAction(
+            jira = jira,
+            meter = meter,
+            jqlMemory = jqlMemory,
+            issueKeyMemory = issueKeyMemory
+        )
+        val searchAStar = SearchJqlStarAction(
+            jira = jira,
+            meter = meter,
+            jqlMemory = jqlMemory,
+            issueKeyMemory = issueKeyMemory
+        )
+        val viewIssue = ViewIssueAction(
+            jira = jira,
+            meter = meter,
+            issueKeyMemory = issueKeyMemory,
+            issueMemory = issueMemory,
+            jqlMemory = jqlMemory
+        )
+        val projectSummary = ProjectSummaryAction(
+            jira = jira,
+            meter = meter,
+            projectMemory = projectMemory
+        )
+        val viewDashboard = ViewDashboardAction(
+            jira = jira,
+            meter = meter
+        )
+        val editIssue = EditIssueAction(
+            jira = jira,
+            meter = meter,
+            issueMemory = issueMemory
+        )
+        val addComment = AddCommentAction(
+            jira = jira,
+            meter = meter,
+            issueMemory = issueMemory
+        )
+        val browseProjects = BrowseProjectsAction(
+            jira = jira,
+            meter = meter,
+            projectMemory = projectMemory
+        )
+
+        val actionProportions = mapOf(
+            createIssue to 1,
+            searchWithJql to 1,
+            searchAStar to 1,
+            viewIssue to 1,
+            projectSummary to 1,
+            viewDashboard to 1,
+            editIssue to 1,
+            addComment to 1,
+            browseProjects to 1
+        )
+
+        actionProportions.entries.forEach { scenario.addMultiple(element = it.key, repeats = it.value) }
+        scenario.shuffle(seededRandom.random)
+        return scenario
+    }
+
+    private fun initializeIssueKeyMemory(seededRandom: SeededRandom) {
+        synchronized(this) {
+            if (this::issueKeyMemory.isInitialized.not()) {
+                issueKeyMemory = AdaptiveIssueKeyMemory(seededRandom)
+            }
+        }
+    }
+
+    /**
+     * You can use `Builder` to share memories between Scenarios.
+     * @since 3.6.0
+     */
+    class Builder {
```

Comment:
This is second place with builder of the same logical purpose. We may consider extraction and refactoring. I mean builder as parametererized type (by proper Scenario implementation class)